File: ruby/testcode/xss_039.rb
require_relative 'shared'

# vuln-code-snippet start ruby_xss_sanitize_allowlist
def render_rich_text(req)
  body = req.param('body')
  # sanitize with strict allowlist — only safe tags/attrs pass through
  clean = sanitize(body, tags: %w[p b i em strong], attributes: %w[class])  # vuln-code-snippet safe-line ruby_xss_sanitize_allowlist
  BenchmarkResponse.html("<article>#{clean}</article>")
end
# vuln-code-snippet end ruby_xss_sanitize_allowlist

File: rust/testcode/xss_039.rs
//! CWE-79: User-supplied value is immediately overwritten with a safe static string before rendering.

// vuln-code-snippet start testcodeXss039
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let mut display = req.param("name");
    display = "Guest".to_string();

    let html = format!("<p>Hello, {}!</p>", display); // vuln-code-snippet target-line testcodeXss039

    super::shared::BenchmarkResponse::ok(&html)
}
// vuln-code-snippet end testcodeXss039
